docs/porting-example/index.rst

```diff
@@ -0,0 +1,345 @@
+  receive `PyObject *` arguments and may still cast them to `PyPointObject *`
+  if they are instances of Point.
+
+* :doc:`steps/step_02_hpy_legacy`: Shows how to transition some methods to HPy
+  methods that receive `HPy` handles as arguments while still supporting legacy
+  methods that receive `PyObject *` arguments.
+
+* :doc:`steps/step_03_hpy_final`: The completed port to HPy where all methods
+  receive `HPy` handles and PyObject_HEAD has been removed.
+
+Take a moment to read through :doc:`steps/step_00_c_api`. Then, once you're
+ready, keep reading.
+
+Each section below corresponds to one of the three porting steps above:
+
+.. contents::
+    :local:
+    :depth: 2
+
+.. note::
+    The steps used here are one approach to porting a module. The specific
+    steps are not required. They're just an example approach.
+
+
+Step 01: Converting the module to a (legacy) HPy module
+-------------------------------------------------------
+
+First for the easy bit -- let's include `hpy.h`:
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: #include <hpy.h>
+    :end-at: #include <hpy.h>
+
+We'd like to differentiate between references to `PyPointObject` that have
+been ported to HPy and those that haven't, so let's rename it to `PointObject`
+and alias `PyPointObject` to `PointObject`. We'll keep `PyPointObject` for
+the instances that haven't been ported yet (the legacy ones) and use
+`PointObject` where we have ported the references:
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: typedef struct {
+    :end-at: } PointObject;
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: typedef PointObject PyPointObject;
+    :end-at: typedef PointObject PyPointObject;
+
+For this step, all references will be to `PyPointObject` -- we'll only start
+porting references in the next step.
+
+Let's also call `HPyType_LEGACY_HELPERS` to define some helper functions
+for use with the `PointObject` struct:
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: HPyType_LEGACY_HELPERS(PointObject)
+    :end-at: HPyType_LEGACY_HELPERS(PointObject)
+
+Again, we won't use these helpers in this step -- we're just setting things
+up for later.
+
+Now for the big steps.
+
+We need to replace `PyType_Spec` for the `Point` type with the equivalent
+`HPyType_Spec`:
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: // HPy type methods and slots (no methods or slots have been ported yet)
+    :end-before: // Legacy module methods (the "dot" method is still a PyCFunction)
+
+Initially the list of ported methods in `point_defines` is empty and all of
+the methods are still in `Point_slots` which we have renamed to
+`Point_legacy_slots` for clarity.
+
+The `PointObject_IS_LEGACY` is defined by the `HPyType_LEGACY_HELPERS` macro
```

Comment:
Since I merged PR #364, this should now be `SHAPE(PointObject)`. I'll take care of rebasing and updating the PR.